library/data/registry-entry-attributes/src/main/java/org/quiltmc/qsl/registry/attribute/api/RegistryEntryAttribute.java

```diff
@@ -0,0 +1,289 @@
+/*
+ * Copyright 2021 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.registry.attribute.api;
+
+import com.mojang.serialization.Codec;
+import org.jetbrains.annotations.Nullable;
+
+import org.quiltmc.qsl.registry.attribute.impl.RegistryEntryAttributeImpl;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+import java.util.Optional;
+import java.util.function.Function;
+
+/**
+ * Represents an attribute that is attached to a registry entry. Maps to a value in a {@link RegistryEntryAttributeHolder}.
+ *
+ * @param <R> registry entry type
+ * @param <V> value type
```

Comment:
These are a little confusing?

`R` is the type of the object in the registry and `V` is the attached data type?

Reply:
Cleared this javadoc up a little.